airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2765,6 +2882,53 @@ def create_simple_retriever(
             transformations=transformations,
             client_side_incremental_sync=client_side_incremental_sync,
         )
+
+        query_properties: Optional[QueryProperties] = None
+        query_properties_key: Optional[str] = None
+        if (
+            hasattr(model.requester, "request_parameters")
+            and model.requester.request_parameters
+            and isinstance(model.requester.request_parameters, Mapping)
+        ):
+            query_properties_definitions = []
+            for key, request_parameter in model.requester.request_parameters.items():
+                if (
+                    isinstance(request_parameter, Mapping)
+                    and request_parameter.get("type") == "QueryProperties"
+                ):
+                    query_properties_key = key
+                    query_properties_definitions.append(request_parameter)
+                elif not isinstance(request_parameter, str):
+                    raise ValueError(
```

Comment:
I couldn't get json schema to pydantic model conversions to enforce `str, `QueryProperties`. It always would get rendered as `Any` regardless of what combo I made once we included both a `#ref` and concrete `type`

Reply:
@brianjlai This is the problem: 

nit: not `$ref"`, but `"$ref"`

here: https://github.com/airbytehq/airbyte-python-cdk/blame/2f29eff3f5d74862e1667f6274a0f23d1b3a6c0c/airbyte_cdk/sources/declarative/declarative_component_schema.yaml#L2036